examples/instanced_shapes/src/main.rs

```diff
@@ -0,0 +1,205 @@
+    let mut instanced_mesh: Gm<InstancedMesh, PhysicalMaterial> = Gm::new(
+        InstancedMesh::new(&context, &instances, &CpuMesh::cube()),
+        PhysicalMaterial::new(
+            &context,
+            &CpuMaterial {
+                albedo: Color {
+                    r: 128,
+                    g: 128,
+                    b: 128,
+                    a: 255,
+                },
+                ..Default::default()
+            },
+        ),
+    );
+
+    // Initial properties of the example, 10 cubes and non instanced.
+    let mut cube_count = 10;
+    let mut is_instanced = false;
+
+    let mut gui = three_d::GUI::new(&context);
+    window.render_loop(move |mut frame_input: FrameInput| {
+        camera.set_viewport(frame_input.viewport);
+
+        // Gui panel to control the number of cubes and whether or not instancing is turned on.
+        let mut panel_width = 0.0;
+        gui.update(
+            &mut frame_input.events,
+            frame_input.accumulated_time,
+            frame_input.viewport,
+            frame_input.device_pixel_ratio,
+            |gui_context| {
+                use three_d::egui::*;
+                SidePanel::left("side_panel").show(gui_context, |ui| {
+                    use three_d::egui::*;
+                    ui.heading("Debug Panel");
+                    ui.add(Slider::new(&mut cube_count, 1..=10000).text("Number of cubes."));
+                    ui.add(Checkbox::new(&mut is_instanced, "Use Instancing"));
+                    ui.add(Label::new(
+                        "Increase the cube count until the cubes don't rotate \
+                                       smoothly anymore, then toggle on instancing. The rotations \
+                                       should become smooth again.",
+                    ));
+                });
+                panel_width = gui_context.used_rect().width() as f64;
+            },
+        );
+
+        // Camera control must be after the gui update.
+        control.handle_events(&mut camera, &mut frame_input.events);
+
+        // Time to move the cubes.
+        let time = (frame_input.accumulated_time * 0.001) as f32;
+
+        // Always update the transforms for both the normal cubes as well as the instanced versions.
+        // This shows that the difference in frame rate is not because of updating the transforms
+        // and shows that the performance difference is not related to how we update the cubes.
+
+        adjust_normal_cubes(&context, time, cube_count, &mut non_instanced_meshes);
+        adjust_instanced_cubes(&context, time, cube_count, &mut instanced_mesh);
+
+        // Then, based on whether or not we render the instanced cubes, collect the renderable
+        // objects.
+        let render_objects: Vec<&dyn Object> = if is_instanced {
+            instanced_mesh.into_iter().collect::<_>()
+        } else {
+            non_instanced_meshes
+                .iter()
+                .map(|x| x as &dyn Object)
```

Comment:
Oh, this reminds me, when I first needed this I came across [this](https://github.com/asny/three-d/discussions/291#discussioncomment-4034372) discussion, I'm not sure how to make that more 'findable' though.

Reply:
I agree, this is not very nice that you have to explicitly convert to `&dyn Object` but the rest of the solution, I quite like. So I really hope that it will convert automatically in some future version of Rust. Until then, I could maybe describe it in the docs of the `render ` functions? Or do you have any other suggestions?

Reply:
Yeah, I think a sample example there would go a long way. I ran into the same problem and found that discussion thread after searching in the repo. The `into_iter()` on the `Gm` was also a bit surprising to me, it's elegant for sure, but surprising if I think about a single `Gm` struct.

Reply:
Yeah, I can see why that would be surprising, if you have any suggestions for improvements, please let me know. I've tried quite a lot of different things and this was the best solution I've tried. I will add some more documentation to the render functions 👍